### DISCUSSION TITLE
More seamless mouse

### DISCUSSION THREAD

**Discussion by Torinde** [dosbox-staging/dosbox-staging#3290] [General]
vbmouse.exe in Staging - you have to get the Staging window in focus _first_ (e.g. by clicking onto it) and only the clicks _afterwards_ propagate to the DOS program (?)

_Originally posted by @Torinde in https://github.com/joncampbell123/dosbox-x/issues/3384#issuecomment-1872487713_

In DOSbox-X with its built-in mouse driver the first click (which gets the window in focus) is also propagated to the DOS program.

**Comment by FeralChild64** [comment]
I did some initial investigation - to see what happens if the window gets focus due to mouse click inside. Unfortunately, on my Linux system, sometimes I am getting SDL_MOUSEBUTTONDOWN event from SDL, and sometimes not.

Could you create a PR? I intend to investigate it, but probably not for 0.81.0.

[edit] Checked with DOSBox-X with seamless mouse enabled - and sometimes the event is passed to the guest side software, sometimes it isn't. Might be OS dependent, window manager dependent, etc.

**Comment by FeralChild64** [comment]
… and on Windows 11 I am never getting the event if window has no focus when clicking inside.

[edit] I think I have a solution, but I'm too tired to test this on Windows today: https://github.com/dosbox-staging/dosbox-staging/tree/fc/seamless-mouse-capture-event-1

**Reply by johnnovak** [reply]
I agree, this would be a nice feature and welcome addition. Common use case (common for me 😄 ) is running a game and a mapping program side by side, so you want to minimise the number of clicks/keypresses needed for "context switching" between playing the game and mapping.

Having to click twice is super annoying. I just haven't ran into this as I'm playing RPGs with keyboard only 90% of the time (at least when I'm in mapping mode).

**Comment by FeralChild64** [comment]
PR created - https://github.com/dosbox-staging/dosbox-staging/pull/3296. Actually, it is enough to provide one hint to SDL library.

**Comment by Torinde** [comment]
Tested the fix, it works, thanks @FeralChild64!
